lib/web_ui/skwasm/surface.cpp

```diff
@@ -28,6 +28,9 @@ Surface::Surface() {
       this);
   // Listen to messages from the worker
   skwasm_registerMessageListener(_thread);
+
+  // Synchronize the time origin for the worker thread
+  skwasm_syncTimeOriginForThread(_thread);
```

Comment:
Could there be a race here? Do we need to "await" for the sync to complete?

Reply:
Nope, it can't race. The `postMessage` queue is serial and all the calls to the web worker go through that queue. Any render calls are guaranteed to occur after the `syncTimeOrigin` message has been processed.